multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -0,0 +1,149 @@
+package org.cloudfoundry.multiapps.mta.resolvers.v2;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.model.ProvidedDependency;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.cloudfoundry.multiapps.mta.resolvers.Reference;
+import org.cloudfoundry.multiapps.mta.resolvers.ReferencePattern;
+
+public class ResourceLiveParameterResolver {
+
+    private static final String CONFIG = "config";
+    private static final String DEFAULT_URL_PLACEHOLDER = "${default-url}";
+    private static final String DEFAULT_URI_PLACEHOLDER = "${default-uri}";
+    private static final String DEFAULT_HOST_PLACEHOLDER = "${default-host}";
+    private static final String DEFAULT_LIVE_URL = "default-live-url";
+    private static final String DEFAULT_LIVE_HOST = "default-live-host";
+    private static final String DEFAULT_LIVE_URI = "default-live-uri";
+    private static final Map<String, String> idleToLiveParameterPairs = Map.of(DEFAULT_URL_PLACEHOLDER, DEFAULT_LIVE_URL,
+                                                                               DEFAULT_URI_PLACEHOLDER, DEFAULT_LIVE_URI,
+                                                                               DEFAULT_HOST_PLACEHOLDER, DEFAULT_LIVE_HOST);
+    private DeploymentDescriptor deploymentDescriptor;
+
+    public ResourceLiveParameterResolver(DeploymentDescriptor deploymentDescriptor) {
+        this.deploymentDescriptor = deploymentDescriptor;
+    }
+
+    public void resolveResource(Resource resource) {
+        if (resource.getParameters()
+                    .containsKey(CONFIG)) {
+            Map<String, Object> serviceCreationParameters = castToMap(resource.getParameters()
+                                                                              .get(CONFIG));
+            serviceCreationParameters.entrySet()
+                                     .forEach(entry -> resolveResourceParameter(entry, resource));
+        }
+    }
+
+    private Map<String, Object> castToMap(Object entryValue) {
+        return (Map<String, Object>) entryValue;
+    }
+
+    private void resolveResourceParameter(Map.Entry<String, Object> entry, Resource resource) {
+        if (entry.getValue() instanceof Map) {
+            resolveMapParameter(castToMap(entry.getValue()), resource);
+        } else if (entry.getValue() instanceof String) {
+            resolveConfigParameter(entry, resource);
+        }
+    }
+
+    private void resolveMapParameter(Map<String, Object> resourceParameter, Resource resource) {
```

Comment:
this doesn't really need to be a separate function, this way the recursion is a bit obfuscated